test/common/http/conn_manager_impl_fuzz_test.cc

```diff
@@ -301,9 +301,16 @@ class FuzzStream {
     }
     case test::common::http::ResponseAction::kHeaders: {
       if (state == StreamState::PendingHeaders) {
-        decoder_filter_->callbacks_->encodeHeaders(
-            std::make_unique<TestHeaderMapImpl>(Fuzz::fromHeaders(response_action.headers())),
-            end_stream);
+        auto headers =
+            std::make_unique<TestHeaderMapImpl>(Fuzz::fromHeaders(response_action.headers()));
+        // The client codec will ensure we always have a valid :status.
```

Comment:
Should we have or point to unit tests which confirm this is the case to regression test?  What if a new codec (say QUIC) screws this up?

Reply:
I've added an ASSERT and docs. In terms of tests, it's a bit tricky we tend to use the codecs themselves to synthesize replies in client tests. So, we could maybe do specific H2 validation with the wire encoding of a status-less HEADERS frame..